### ISSUE TITLE
TypeAlias makes forward reference Unbound

### ISSUE THREAD

**Issue by iyakushev** [microsoft/pyright#2255]
**Describe the bug**
Whenever I use `TypeAlias` any attempt at forward referencing some type makes it `Unbound`. But, if I delete explicit `TypeAlias` pyright produces a correct type signature.

_Edit_: I can confirm that python's 3.10 `typing.TypeAlias` has the same behavior.

**To Reproduce**
```python
from typing import Union
from typing_extensions import TypeAlias

 # correct type. `Type[list[str | RecursiveType]]
RecursiveType = list[Union[str, "RecursiveType"]]

# makes RecursiveType reference unbound. `Type[list[str | Unbound]]`
RecursiveType: TypeAlias = list[Union[str, "RecursiveType"]] 
```
**Expected behavior**
I believe that explicit `TypeAlias` should not interfere with forward references. Although, I might've missed something and will be happy to be proven wrong. Thank you in advance.

**VS Code extension or command-line**
pyright 1.1.164
python 3.9.4

**Comment by erictraut** [comment]
Thanks for the bug report. This appears to be specific to recursive type aliases, not forward references in general. I'll look into it.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.166, which I just published. It will also be included in the next release of pylance.
